fdbclient/Metacluster.cpp

```diff
@@ -131,9 +133,9 @@ std::string MetaclusterTenantMapEntry::toJson() const {
 	return json_spirit::write_string(json_spirit::mValue(tenantEntry));
 }
 
-bool MetaclusterTenantMapEntry::matchesConfiguration(MetaclusterTenantMapEntry const& other) const {
+bool MetaclusterTenantMapEntry::matchesConfiguration(MetaclusterTenantMapEntry const& other, bool matchAssigned) const {
 	return tenantName == other.tenantName && tenantLockState == other.tenantLockState &&
-	       tenantGroup == other.tenantGroup && assignedCluster == other.assignedCluster;
+	       tenantGroup == other.tenantGroup && (!matchAssigned || assignedCluster == other.assignedCluster);
```

Comment:
We may just want this to compare only tenant groups for now. The idea here is that we are checking whether the configurable properties of a tenant match what is existing, and although the assigned cluster is quasi-configurable, I think that we are checking that elsewhere. The metacluster doesn't really track lock state yet, and the name is also considered as a separate thing.